.github/workflows/codespell.yml

```diff
@@ -0,0 +1,25 @@
+# Codespell configuration is within .codespellrc
+---
+name: Codespell
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Annotate locations with typos
+        uses: codespell-project/codespell-problem-matcher@v1
```

Comment:
Hmm, this does not appear to have seen much interest or activity:

https://github.com/codespell-project/codespell-problem-matcher

Reply:
Yep, but it seems just to work in my experience

Reply:
Right, I trust that it works today. What I have less confidence in is a bad actor coming in, somehow gaining control of `codespell-problem-matcher`, changing the `v1` tag to point at something else, and anyone noticing.

Reply:
yeap, there was a similar concern/practice elsewhere recently (https://github.com/argoproj/argo-workflows/pull/14454) -- I  will change now to current commit hashes...  ran `pin-github-action .github/workflows/codespell.yml` and pushed